packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -580,9 +591,12 @@ class FloatingActionButton extends StatelessWidget {
       ?? theme.materialTapTargetSize;
     final bool enableFeedback = this.enableFeedback
       ?? floatingActionButtonTheme.enableFeedback ?? true;
-    final TextStyle textStyle = theme.textTheme.button!.copyWith(
+    final TextStyle extendedTextStyle = (this.extendedTextStyle
```

Comment:
This is a bit hard to read. Here's a suggestion:
```dart
final TextStyle extendedTextStyle = (this.extendedTextStyle
  ?? floatingActionButtonTheme.extendedTextStyle
  ?? theme.textTheme.button!.copyWith(letterSpacing: 1.2)).copyWith(color: foregroundColor);
```

Reply:
Done